### PR TITLE
Fixed sending of notifications.

### PR DESCRIPTION
**Pull request by BSolonenko**
This PR is **[ready]** for review.

### Risk
This PR makes **[no]** API changes.

### Summary
- Fixed RCRPCPlugin::OnPolicyEvent;
 - Fixed RCRPCPlugin::OnApplicationEvent;
 - Fixed OnExitApplicationNotification::Run;
 - Fixed ApplicationManagerImpl::UnregisterApplication;
 - Fixed ApplicationManagerImpl::OnPTUFinished;
 - Refactoring ApplicationImpl::RemoveExtension;
 - is_remote_control_supported_ added  in initialization list.

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/application_manager/src/application_manager_impl.cc [hunk 1]
```diff
@@ -2442,6 +2442,15 @@ void ApplicationManagerImpl::UnregisterApplication(
     MessageHelper::SendStopAudioPathThru(*this);
   }
 
+#ifdef SDL_REMOTE_CONTROL
+  std::function<void(plugin_manager::RPCPlugin&)> on_app_unregistered =
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/src/application_manager_impl.cc]: @BSolonenko actually, you could replace this long type declaration with `auto` as you did in all other places

**Comment by LitvinenkoIra** [comment]
@BSolonenko please fix UTs

**Comment by BSolonenko** [comment]
@LitvinenkoIra  UTs fixed.

**Comment by BSolonenko** [comment]
@AByzhynar  Please review.

**Pull request merged**